model/task/task_annotations.go

```diff
@@ -0,0 +1,195 @@
+package task
+
+import (
+	"time"
+
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/evergreen/model/annotations"
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+// MoveIssueToSuspectedIssue removes an issue from an existing annotation and adds it to its suspected issues,
+// and unsets its associated task document as having annotations if this was the last issue removed from the
+// annotation.
+func MoveIssueToSuspectedIssue(taskId string, taskExecution int, issue annotations.IssueLink, username string) error {
+	newIssue := issue
+	newIssue.Source = &annotations.Source{Requester: annotations.UIRequester, Author: username, Time: time.Now()}
+	q := annotations.ByTaskIdAndExecution(taskId, taskExecution)
+	q[bsonutil.GetDottedKeyName(annotations.IssuesKey, annotations.IssueLinkIssueKey)] = issue.IssueKey
+	if err := db.Update(
```

Comment:
Have you considered using FindAndModify for this (and for RemoveIssue) instead of update and then a subsequent find? It combines those calls and makes the updates a bit more atomic.

Reply:
Sounds good, I added it for those two functions.